backend/src/statistics/tests/statistics.e2e.spec.ts

```diff
@@ -0,0 +1,240 @@
+
+  beforeEach(async () => {
+    await client.reset();
+    await client.category.create({
+      data: {
+        id: 1,
+        title: "Algorithms",
+      },
+    });
+    await client.topic.createMany({
+      data: [
+        {
+          id: 1,
+          name: "Dynamic Programming",
+        },
+        {
+          id: 2,
+          name: "Bit Manipulation",
+        },
+        {
+          id: 3,
+          name: "Greedy",
+        },
+        {
+          id: 4,
+          name: "Memoization",
+        },
+        {
+          id: 5,
+          name: "Two Pointers",
+        },
+        {
+          id: 6,
+          name: "String",
+        },
+      ],
+    });
+    await client.question.create({
+      data: {
+        id: 1,
+        title: "Integer Replacement",
+        difficulty: Difficulty.MEDIUM,
+        categoryId: 1,
+        description: "description",
+        topics: {
+          connect: [{ id: 1 }, { id: 2 }, { id: 3 }, { id: 4 }],
+        },
+      },
+    });
+    await client.question.create({
+      data: {
+        id: 2,
+        title: "Remove Palindromic Subsequences",
+        difficulty: Difficulty.EASY,
+        categoryId: 1,
+        description: "description",
+        topics: {
+          connect: [{ id: 5 }, { id: 6 }],
+        },
+      },
+    });
+    await client.roomSession.create({
+      data: {
+        id: "1",
+        startTime: startTimeRoomOne,
+        endTime: endTimeRoomOne,
+        questionId: 1,
+      },
+    });
+    await client.roomSession.create({
+      data: {
+        id: "2",
+        startTime: startTimeRoomTwo,
+        endTime: endTimeRoomTwo,
+        questionId: 2,
+      },
+    });
+    await client.submission.create({
+      data: {
+        id: "1",
+        roomSessionId: "1",
+        status: Status.ACCEPTED,
+        language: Language.CPP,
+        expectedOutput: "expectedOutput",
+        code: "code",
+        createdAt: endTimeRoomOne,
+        updatedAt: endTimeRoomOne,
+      },
+    });
+    await client.submission.create({
+      data: {
+        id: "2",
+        roomSessionId: "2",
+        status: Status.PENDING,
+        language: Language.PYTHON,
+        expectedOutput: "expectedOutput",
+        code: "code",
+        createdAt: endTimeRoomTwo,
+        updatedAt: endTimeRoomTwo,
+      },
+    });
+    await client.user.create({
+      data: {
+        id: 1,
+        email: "janedoe@email.com",
+        name: "Jane Doe",
+        password: "password",
+        roomSessions: {
+          connect: [{ id: "1" }, { id: "2" }],
+        },
+      },
+    });
+    await client.user.create({
+      data: {
+        id: 2,
+        email: "johndoe@email.com",
+        name: "John Doe",
+        password: "password",
+        roomSessions: {
+          connect: [{ id: "1" }, { id: "2" }],
+        },
+      },
+    });
+  });
+
+  describe("GET /user/statistics", () => {
+    it(`get default statistics`, async () => {
+      const expected = {
```

Comment:
As an aside, since there is no guaranteed order to the arrays received, it's generally best to sort them before comparing them. But for now we can skip over this.